src/run.ts

```diff
@@ -0,0 +1,121 @@
+import * as core from '@actions/core';
+import { load } from "js-yaml";
+import { readFileSync } from "fs";
+import { throttling } from '@octokit/plugin-throttling';
+import { GitHub, getOctokitOptions } from '@actions/github/lib/utils'
+type Octokit = InstanceType<typeof GitHub>;
+
+interface Input {
+  token: string;
+  org: string;
+  file: string;
+  forceEnable: boolean;
+}
+
+export function getInputs(): Input {
+  const result = {} as Input;
+  result.token = core.getInput('github-token');
+  result.org = core.getInput('org');
+  result.file = core.getInput('file');
+  result.forceEnable = core.getBooleanInput('force-enable');
+  return result;
+}
+
+const getRepoNames = async (octokit: Octokit, orgLogin: string): Promise<string[]> => {
+  let repoNames: string[] = [];
+  let _hasNextPage = true;
+  let _endCursor = null;
+  while (_hasNextPage) {
+    const {
+      organization: {
+        repositories: {
+          nodes: repositories,
+          pageInfo: {
+            hasNextPage,
+            endCursor
+          }
+        }
+      }
+    } = await octokit.graphql(`{ 
+      organization(login:"${orgLogin}") {
+        repositories(first:100, after:${JSON.stringify(_endCursor)}) {
+          nodes {
+            name
+          }
+          pageInfo {
+            hasNextPage
+            endCursor
+          }
+        }
+      }
+    }`);
+    _hasNextPage = hasNextPage;
+    _endCursor = endCursor;
+    const names: string[] = repositories
+      .map(repo => repo.name)
+      .filter(name => name !== orgLogin);
+    core.info(names.join('\n'));
+    repoNames = repoNames.concat(names);
+  }
+  return repoNames;
+}
+
+const createOctokit = (token: string): Octokit => {
+  return new (GitHub.plugin(throttling))({
+    ...getOctokitOptions(token),
+    throttle: {
+      onRateLimit: (retryAfter, options, octokit) => {
+        octokit.log.warn(`Request quota exhausted for request ${options.method} ${options.url}`);
+        if (options.request.retryCount === 0) {
+          octokit.log.info(`Retrying after ${retryAfter} seconds!`);
+          return true;
+        }
+        return false;
+      },
+      onSecondaryRateLimit: (_, options, octokit) => {
+        octokit.log.warn(`SecondaryRateLimit detected for request ${options.method} ${options.url}`);
+      },
+    }
+  });
+}
+
+const run = async (): Promise<void> => {
+  try {
+    const input = getInputs();
+
+    const reposAllowed = {};
```

Comment:
Similar to my above comment, try and be a bit more strict with your typing here 🙇 what data do you expect to be in this object? Maybe create a custom type here. Similar to how you have done:

```
  const result = {} as Input;
```

Reply:
The function `getInputs()` returns type `Input`. Seems that the type is inferred by the function call and I don't need to specify it. I was considering it.
<img width="278" alt="Screen Shot 2022-05-31 at 2 39 19 PM" src="https://user-images.githubusercontent.com/22425467/171261414-73ce30df-398c-4a30-b14c-7f6436839108.png">

Reply:
Sorry @austenstone, this suggested changes didn't do my any favours 😢 

I actually meant `const reposAllowed = {};`, I used `  const result = {} as Input;` as a good example. I think `const reposAllowed = {};` could use with stricter typing 🙇